src/engines/nwn2/roster.cpp

```diff
@@ -34,29 +36,44 @@ namespace Engines {
 
 namespace NWN2 {
 
-Roster::Roster() : _lastRetrieved(UINT32_MAX) {
+Roster::Roster() : _partyLimit(3), _lastRetrieved(UINT32_MAX) {
 	load();
 }
 
 Roster::~Roster() {
 }
 
-/** Add a creature by template name to the roster list. */
-bool Roster::addRosterMemberByTemplate(Common::UString name, Common::UString cTemplate) {
+uint32 Roster::getRosterNPCPartyLimit() {
+	return _partyLimit;
+}
+
+void Roster::setRosterNPCPartyLimit(uint32 limit) {
+	_partyLimit = limit;
+}
+
+/**
+ * Add a creature by template name to the roster list.
+ * The 'unitTest' flag is required for unit testing
+ * because the UTC files will not be available.
+ */
+bool Roster::addRosterMemberByTemplate(Common::UString name, Common::UString cTemplate, bool unitTest) {
 	// Disallow null strings
 	if (name.size() == 0 || cTemplate.size() == 0)
 		return false;
 
 	// Don't permit duplicate roster names
-	if (getRosterMember(name) != _members.end())
+	auto matchName = [&name](const Member &m) bool {return m.rosterName == name;};
```

Comment:
Return type is done with `->`. See (2) here: https://en.cppreference.com/w/cpp/language/lambda